python_backend/langchain_pipeline/chains/rag_chain.py

```
@@ -89,6 +74,45 @@ def __init__(self, model_name: str = "gpt-4", temperature: float = 0.7):
         # 자동 초기화 시도
         self._load_vectorstore()
     
+    def _check_services_availability(self):
+        """Services 가용성 확인 (실제 로드하지 않음)"""
+        try:
+            import services.prompt_engineering
+            import services.openai_services
+            import services.conversion_service
+            import services.user_preferences
```

Ruff (0.12.2)

`services.user_preferences` imported but unused; consider using `importlib.util.find_spec` to test for availability

(F401)